src/library/playlistfeature.cpp

```diff
@@ -177,6 +177,41 @@ void PlaylistFeature::buildPlaylistList() {
     }
 }
 
+void PlaylistFeature::updatePlaylistList(int playlist_id) {
+    // Setup the sidebar playlist model
+    QSqlTableModel playlistTableModel(this, m_pTrackCollection->database());
+    playlistTableModel.setTable("PlaylistsCountsDurations");
+    QString filter = "id=" + QString::number(playlist_id);
+    playlistTableModel.setFilter(filter);
+    playlistTableModel.select();
+    while (playlistTableModel.canFetchMore()) {
+        playlistTableModel.fetchMore();
+    }
+    QSqlRecord record = playlistTableModel.record();
+    int nameColumn = record.indexOf("name");
+    int countColumn = record.indexOf("count");
+    int durationColumn = record.indexOf("durationSeconds");
+
+    for (int row = 0; row < playlistTableModel.rowCount(); ++row) {
+        QString name = playlistTableModel.data(
+                playlistTableModel.index(row, nameColumn)).toString();
+        int count = playlistTableModel.data(
+                playlistTableModel.index(row, countColumn)).toInt();
+        int duration = playlistTableModel.data(
+                playlistTableModel.index(row, durationColumn)).toInt();
+        for (auto it = m_playlistList.begin();
+                it != m_playlistList.end(); ++it) {
+            if (it->first == playlist_id) {
+                it->second = QString("%1 (%2) %3").arg(name,
+                        QString::number(count),
+                        mixxx::Duration::formatSeconds(duration));
+                break;
+            }
+        }
+        break;
```

Comment:
This is a very strange way to fetch just the first element from the result set. It took me some time until I noticed that the outer loop is actually not a loop at all.

Reply:
Yes, I think it is my leak of knowledge how to use the Qt API correct in this case. I we haves such constructs in other places where we loop through a table even though we expect only one or no rows. How can I fix this?

Reply:
We expect at most a single row here. This can easily be expressed through the code.

```
DEBUG_ASSERT(playlistTableModel.rowCount() <= 1);
if (playlistTableModel.rowCount() > 0) {
...
```

Reply:
Ah, yes that's easy. Can we also fix the while loop above?